src/callbacks/get_callbacks.jl

```diff
@@ -40,9 +41,26 @@ function get_diagnostics(
         num_netcdf_points =
             tuple(parsed_args["netcdf_interpolation_num_points"]...)
     else
-        # TODO: Once https://github.com/CliMA/ClimaCore.jl/pull/1567 is merged,
-        # dispatch over the Grid type
-        num_netcdf_points = (180, 90, 50)
+        # Estimate the number of points we need to cover the entire domain
+        # ncolumns is the number of local columns
+        tot_num_columns =
+            ClimaComms.nprocs(context) * Fields.ncolumns(axes(Y.c))
+        if parsed_args["config"] == "plane"
+            num1, num2 = tot_num_columns, 0
+        elseif parsed_args["config"] == "sphere"
+            num2 = round(Int, sqrt(tot_num_columns / 2))
+            num1 = 2num2
+        elseif parsed_args["config"] == "box"
+            num2 = round(Int, sqrt(tot_num_columns))
+            num1 = num2
+        elseif parsed_args["config"] == "column"
+            # We need at least two points horizontally because our column is
+            # actually a box
+            num1, num2 = 2, 2
+        else
+            error("Uncaught case")
+        end
```

Comment:
I just want to point out that every time we used `parsed_args` for if-else statements like this (instead of using, e.g., the `atmos` struct), we may be introducing more inference failures, because the compiler may not be able to guarantee that this won't result in an error.